### PR TITLE
Add codegen for elm-pages, allow picking which effects to generate

### PR DESCRIPTION
**Pull request by miniBill**

### CODE REVIEW

File: src/Cli.elm [hunk 1]
```diff
@@ -70,7 +76,15 @@ program =
...
   --module-name                      The Elm module name. Default to <OAS info.title>
-    
+
+  --effect-types                     Which kind of APIs to generate, the options are:
+                                      - cmd: input -> Cmd msg
+                                      - riskycmd: as above, but using Http.riskyRequest
+                                      - task: input -> Task Http.Error msg
+                                      - riskytask: as above, but using Http.riskyRequest
+                                      - backendtask: for dillonkearns/elm-pages
+                                      - concurrenttask: for andrewMacmurray/elm-concurrent-task
```
Comment (wolfadex) [inline src/Cli.elm:86]: Nice! ❤️

File: src/Cli.elm [hunk 2]
```diff
@@ -93,6 +107,41 @@ program =
...
+            Ok Task
+
+        "taskrisky" ->
+            Ok TaskRisky
+
+        "backendtask" ->
+            Ok BackendTask
+
+        _ ->
+            Err <| "Unexpected effect type: " ++ effectType
```
Comment (wolfadex) [inline src/Cli.elm:142]: This is reminding me of the horrors of a CLI at Square written in Go that didn't handle bad values

**Comment by miniBill** [comment]
I actually haven't implemented the concurrent task part yet, I should
probably comment it out of the docs

Il mar 28 mag 2024, 03:54 Wolfgang Schuster ***@***.***> ha
scritto:

File: src/Cli.elm (hunk 2 above)
Comment (miniBill) [reply src/Cli.elm:142]: Oh dear.
I really appreciate that Elm forces us to handle all cases

File: src/Cli.elm (hunk 1 above)
Comment (miniBill) [reply src/Cli.elm:86]: I actually haven't implemented the concurrent task part yet, I should
probably comment it out of the docs

**Review by wolfadex — APPROVED** [review]
I think it looks good overall. Just the elm-review cleanup and the concurrent task thing you mentioned.

**Pull request merged**